cpp/src/arrow/flight/sql/client.h

```diff
@@ -408,7 +408,6 @@ class ARROW_FLIGHT_SQL_EXPORT FlightSqlClient {
   /// \brief Explicitly shut down and clean up the client.
   Status Close();
 
- protected:
```

Comment:
Do we need this?

Reply:
Yes. I have to make these public to make the local (to the .cc file) helper able to call these.

More details here: https://github.com/apache/arrow/pull/36559/files#r1693705438

Reply:
I see.